### PR TITLE
Use API for switching pull-request base

### PR DESCRIPTION
**Pull request by sstok**
| Q | A |
| --- | --- |
| Fixed tickets |  |
| License | MIT |

GitHub has added support to allow switching the base of a pull-request (after creation!).
So no need to create a new pull-request anymore.

https://github.com/blog/2224-change-the-base-branch-of-a-pull-request

### CODE REVIEW

File: src/Command/PullRequest/PullRequestSwitchBaseCommand.php [hunk 1]
```diff
@@ -34,9 +34,8 @@ protected function configure()
             ->addArgument('pr_number', InputArgument::REQUIRED, 'PR number to be switched')
             ->addArgument(
                 'base_branch',
-                InputArgument::OPTIONAL,
-                'Name of the new base branch to switch the PR to',
-                'master'
+                InputArgument::REQUIRED,
+                'Name of the new base branch to switch the PR to'
```
Comment (sstok) [inline src/Command/PullRequest/PullRequestSwitchBaseCommand.php]: The base_branch argument is now required, but this seemed to be a bug as you don't want to auto switch to master always. I a follow-up pr we may consider making this optional and showing a selection menu when missing 👍

File: src/Command/PullRequest/PullRequestSwitchBaseCommand.php [hunk 2]
```diff
@@ -96,26 +95,41 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
-
-        $switchPr = $adapter->switchPullRequestBase(
-            $prNumber,
-            $baseBranch,
-            $sourceOrg.':'.$branchName.'-switched',
-            $input->getOption('force-new-pr')
-        );
 
-        if ($prNumber === $switchPr['number']) {
-            $adapter->createComment($prNumber, sprintf('(PR base switched to %s)', $baseBranch));
```
Comment (sstok) [inline src/Command/PullRequest/PullRequestSwitchBaseCommand.php]: GitHub already informs about this, other adapters can add a comment manually.
All though I'm not sure if the other adapters are fully implemented at the moment.

**Comment by cordoval** [comment]
this changes src/Command/PullRequest/PullRequestSwitchBaseCommand.php does this mean it leaks implementation details from the specific adapter into the command? In addition what happens with the other adapters, will they also be required to support the new feature of github?

**Comment by sstok** [comment]
Other adapters simple close and create a new pull-request. 

BitBucket supports to change the base branch (not sure why I did not implement this), GitLab does to! GitHub was the only one who did not support this, I will update the other adapters as well 👍

**Comment by sstok** [comment]
I will try to finish this asap, sorry for the delay.

**Pull request closed without merging**